cli/src/opts/wallet.rs

```diff
@@ -12,25 +13,25 @@ use serde::Serialize;
 
 #[derive(Debug)]
 pub enum WalletType {
-    Local(SignerMiddleware<Provider<Http>, LocalWallet>),
-    Ledger(SignerMiddleware<Provider<Http>, Ledger>),
-    Trezor(SignerMiddleware<Provider<Http>, Trezor>),
+    Local(SignerMiddleware<Arc<Provider<RetryClient<Http>>>, LocalWallet>),
+    Ledger(SignerMiddleware<Arc<Provider<RetryClient<Http>>>, Ledger>),
+    Trezor(SignerMiddleware<Arc<Provider<RetryClient<Http>>>, Trezor>),
 }
```

Comment:
Meh this is such an ugly signature, we should upstream that somehow to ethers so that we don't need to think about these types again

Reply:
I think they can all go away when the object safe PR goes through